docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -381,23 +408,46 @@ spec:
 
 * Infrastructure templates are expected to be immutable, so infrastructure template contents do not have to hashed in order to detect
   changes.
-   
+
+##### KubeadmControlPlane rollout (using delete-and-create)
+
+KCP rollout delete-and-create is recommended for resource constrained environment like bare-metal, OpenStack or vSphere resource pools, etc.
+
+- Triggered by:
+    - Changes to Version
+    - Changes to the kubeadmConfigSpec
+    - Changes to the infrastructureRef
+    - `KubeadmControlPlane.Spec.UpgradeStrategy` is set
+
+- Rollout operations rely on scale up and scale down which are be blocked based on Etcd and control plane health checks.
```

Comment:
I think it is worth to document health checks here, otherwise it is not clear how the scale down protect the cluster from potential problems that can happen when deleting a machines